Pilot/pilotCommands.py

```diff
@@ -1100,3 +1101,111 @@ def execute( self ):
     self.__startJobAgent()
 
     sys.exit( 0 )
+
+class NagiosProbes( CommandBase ):
+  """ Run one or more Nagios probe scripts that follow the Nagios Plugin API:
+       https://assets.nagios.com/downloads/nagioscore/docs/nagioscore/3/en/pluginapi.html
+  
+      Each probe is a script or executable present in the pilot directory, which is
+      executed to gather its return code and stdout messages. Probe name = filename.
+      
+      Probes must not expect any command line arguments but can gather information about
+      the current machine from expected environment variables etc.
+  
+      The results are reported through the Pilot Logger.
+  """
+
+  def __init__( self, pilotParams ):
+    """ c'tor
+    """
+    super( NagiosProbes, self ).__init__( pilotParams )
+    self.nagiosProbes = []
+    self.nagiosPutURL = None
+
+  def _setNagiosOptions( self ):
+    """ Setup list of Nagios probes and optional PUT URL from pilot.json
+    """
+   
+    try:
+      self.nagiosProbes = [str( pv ).strip() for pv in self.pp.pilotJSON['Setups'][self.pp.setup]['NagiosProbes'].split(',')]
+    except KeyError:
+      try:
+        self.nagiosProbes = [str( pv ).strip() for pv in self.pp.pilotJSON['Setups']['Defaults']['NagiosProbes'].split(',')]
+      except KeyError:
+        pass
+
+    try:
+      self.nagiosPutURL = str( self.pp.pilotJSON['Setups'][self.pp.setup]['NagiosPutURL'] )
+    except KeyError:
+      try:
+        self.nagiosPutURL = str( self.pp.pilotJSON['Setups']['Defaults']['NagiosPutURL'] )
+      except KeyError:
+        pass
+
+    self.log.debug( 'NAGIOS PROBES [%s]' % ', '.join( self.nagiosProbes ) )
+
+  def _runNagiosProbes( self ):
+    """ Run the probes one by one
+    """
+
+    for probeCmd in self.nagiosProbes:
+     self.log.debug( "Running Nagios probe %s" % probeCmd )
```

Comment:
The indentation here is wrong. Please run pylint or follow what is in https://landscape.io/github/DIRACGrid/Pilot/pulls/21